novawallet/Common/ViewController/CardLayoutPresentationController/Transitions/CardPresentingTransition.swift

```diff
@@ -0,0 +1,152 @@
+            translationX: 0,
+            y: finalFrame.minY
+        )
+    }
+
+    func createDimmingView(for view: UIView) -> UIView {
+        let dimmingView = UIView(frame: view.bounds)
+        let dimmingColor = UIColor.white.withAlphaComponent(0.1)
+
+        dimmingView.backgroundColor = dimmingColor
+        dimmingView.clipsToBounds = true
+        dimmingView.tag = dimmingViewTag
+
+        return dimmingView
+    }
+
+    func calculateFinalDestinationFrame(using sourceView: UIView) -> CGRect {
+        let finalOrigin = sourceView.bounds.offsetBy(
+            dx: 0,
+            dy: CardLayoutPresentationController.topOffset()
+        ).origin
+
+        let finalFrameSize = CGSize(
+            width: sourceView.bounds.width,
+            height: sourceView.bounds.height - CardLayoutPresentationController.topOffset()
+        )
+
+        let finalDestinationViewFrame = CGRect(
+            origin: finalOrigin,
+            size: finalFrameSize
+        )
+
+        return finalDestinationViewFrame
+    }
+}
+
+// MARK: UIViewControllerAnimatedTransitioning
+
+extension CardPresentingTransition: UIViewControllerAnimatedTransitioning {
+    func transitionDuration(using _: UIViewControllerContextTransitioning?) -> TimeInterval {
+        transitionDuration
+    }
+
+    func animateTransition(using transitionContext: UIViewControllerContextTransitioning) {
+        guard
+            let sourceController = transitionContext.viewController(forKey: .from),
+            let destinationController = transitionContext.viewController(forKey: .to)
+        else { return }
+
+        let coveredContextView: UIView
+
+        if let tabBarController = sourceController as? UITabBarController {
+            let navController = tabBarController.selectedViewController as? UINavigationController
+            coveredContextView = navController?.topViewController?.view ?? tabBarController.view
+        } else {
+            coveredContextView = sourceController.view
+        }
+
+        let initialDestinationViewFrame = coveredContextView.bounds.offsetBy(
+            dx: 0,
+            dy: coveredContextView.bounds.maxY
+        )
+        let finalDestinationViewFrame = calculateFinalDestinationFrame(using: coveredContextView)
+
+        let destinationTransform = createDestinationViewTransform(finalDestinationViewFrame)
+        let sourceTransform = createSourceViewTransform(sourceController.view)
+
+        coveredContextView.layer.masksToBounds = true
+        destinationController.view.layer.masksToBounds = true
+        destinationController.view.frame = initialDestinationViewFrame
+
+        transitionContext.containerView.addSubview(destinationController.view)
+
+        let dimmingView = createDimmingView(for: coveredContextView)
+        dimmingView.alpha = 0
+
+        coveredContextView.addSubview(dimmingView)
+
+        // TODO: Refactor to using BlockViewAnimator after UIKit update to support spring params
```

Comment:
I think SpringAnimator already should have all needed params

Reply:
Will introduce new animator